File: modules/admin-ui/src/main/resources/web/admin/common/js/content/site/Site.ts
module api.content.site {

    import Property = api.data.Property;
    import PropertyIdProvider = api.data.PropertyIdProvider;
    import ApplicationKey = api.application.ApplicationKey;
    import ValueTypes = api.data.ValueTypes;

    export class Site extends api.content.Content implements api.Equitable, api.Cloneable {

        constructor(builder: SiteBuilder) {
            super(builder);
        }

        isSite(): boolean {
            return true;
        }

        getDescription(): string {
            return this.getContentData().getString("description");
        }

        getSiteConfigs(): SiteConfig[] {

            var siteConfigs: SiteConfig[] = [];
            this.getContentData().forEachProperty("siteConfig", (moduleProperty: Property) => {
                var siteConfigData = moduleProperty.getPropertySet();
                if (siteConfigData) {
                    var siteConfig = SiteConfig.create().fromData(siteConfigData).build();
                    siteConfigs.push(siteConfig);
                }
            });

            return siteConfigs;
        }

        getApplicationKeys(): ApplicationKey[] {
            return this.getSiteConfigs().map((config: SiteConfig) => config.getApplicationKey());
        }

        equals(o: api.Equitable, ignoreEmptyValues: boolean = false): boolean {

            if (!api.ObjectHelper.iFrameSafeInstanceOf(o, Site)) {
                return false;
            }

            return super.equals(o, ignoreEmptyValues);
        }

        clone(): Site {

            return this.newBuilder().build();
        }

        newBuilder(): SiteBuilder {
            return new SiteBuilder(this);
        }
    }

    export class SiteBuilder extends api.content.ContentBuilder {

        constructor(source?: Site) {
            super(source);
        }

        fromContentJson(contentJson: api.content.json.ContentJson, propertyIdProvider: PropertyIdProvider): SiteBuilder {
            super.fromContentJson(contentJson, propertyIdProvider);
            return this;
        }

        build(): Site {
            return new Site(this);
        }
    }
}

File: modules/core/core-api/src/main/java/com/enonic/xp/site/Site.java
package com.enonic.xp.site;

import com.enonic.xp.annotation.PublicApi;
import com.enonic.xp.content.Content;
import com.enonic.xp.data.PropertyTree;
import com.enonic.xp.schema.content.ContentTypeName;

@PublicApi
public final class Site
    extends Content
{
    private Site( final Builder builder )
    {
        super( builder );
    }

    public String getDescription()
    {
        return this.getData().getString( "description" );
    }

    @Override
    public boolean equals( final Object o )
    {
        if ( this == o )
        {
            return true;
        }
        if ( o == null || getClass() != o.getClass() )
        {
            return false;
        }

        return super.equals( o );
    }

    public static Builder create()
    {
        return new Builder();
    }

    public static Builder create( final Site source )
    {
        return new Builder( source );
    }

    public static final class Builder
        extends Content.Builder<Builder>
    {
        public Builder( final Site source )
        {
            super( source );
        }

        public Builder()
        {
            super();
            type = ContentTypeName.site();
        }

        public Builder description( final String description )
        {

            if ( data == null )
            {
                data = new PropertyTree();
            }
            data.setString( "description", description );
            return this;
        }

        @Override
        public Site build()
        {
            return new Site( this );
        }

    }
}
